app/Http/Controllers/Api/AssetsController.php

```diff
@@ -716,16 +718,21 @@ public function update(ImageUploadRequest $request, $id)
                 }
             }
 
+            if (($request->filled('assigned_user')) || ($request->filled('assigned_asset')) || ($request->filled('assigned_location'))) {
+                app('App\Http\Requests\AssetCheckoutRequest');
```

Comment:
I haven't given this a full look yet but this is really standing out to me. I think calling a request in this way and not in the expected "parameter in a controller method" is, for me, jarring and not in the spirit how they should be used 😅

Reply:
I opted to do it this way to consider PATCH *and* POST, so we skip the whole validation altogether if the request doesn't have the field or the field is empty.

Open to hearing alternate suggestions tho.

Reply:
Wait @marcusmoore I think I understand what you're saying - you're saying - 

Instead of saying:

`$request->filled('assigned_user')`

We should be saying:

`$this->assigned_user`

To fit with the rest of the whole "FormRequest ethos"? If that's all that's no big deal and, yes, you're right.

Reply:
We invoke FormRequests independently using the app facade elsewhere in the code. (See the image/file upload requests sprinkled throughout), so it's not like a one-time thing. 90% of the time, the update is just an update. Sometimes it's not and it's also a checkout.

Reply:
@uberbrady I think he's talking about the method signature itself and the manual invocation of the form request (so we can use more than one form request within a controller) - but I could be wrong :)

Reply:
> I opted to do it this way to consider PATCH and POST, so we skip the whole validation altogether if the request doesn't have the field or the field is empty.

Can we introduce that logic to the form request itself? On a similar note, I'm not all that adverse to the duplication that comes with having dedicated "Store" and "Update" form requests because of how often we end up with different rules for each operation but that's a bigger topic for another time. 

> We invoke FormRequests independently using the app facade elsewhere in the code. (See the image/file upload requests sprinkled throughout), so it's not like a one-time thing. 90% of the time, the update is just an update. Sometimes it's not and it's also a checkout.

But it still goes against the traditional way form requests are invoked and intended to be used. I'm all for finding unique ways to play with the tools we're provided but I'm hesitant/side-eyeing this one...